packages/core/src/checkout-buttons/strategies/braintree/braintree-paypal-button-strategy.ts

```diff
@@ -137,16 +158,25 @@ export default class BraintreePaypalButtonStrategy implements CheckoutButtonStra
 
     private async _setupPayment(
         braintreePaypalCheckout: BraintreePaypalCheckout,
-        shippingAddress?: Address | null,
-        onError?: (error: BraintreeError | StandardError) => void
+        braintreepaypal: BraintreePaypalButtonInitializeOptions
     ): Promise<string> {
-        const state = await this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout());
+        const { buyNowInitializeOptions, shippingAddress, onPaymentError } = braintreepaypal;
+        let state: InternalCheckoutSelectors;
 
         try {
-            const checkout = state.checkout.getCheckoutOrThrow();
-            const config = state.config.getStoreConfigOrThrow();
+            this._buyNowCart = await this._createBuyNowCart({ buyNowInitializeOptions });
+
+            if (this._buyNowCart) {
+                state = this._store.getState();
+            } else {
+                state = await this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout());
+            }
+
             const customer = state.customer.getCustomer();
 
+            const amount = this._buyNowCart ? this._buyNowCart?.cartAmount : state.checkout.getCheckoutOrThrow().outstandingBalance;
+            const currencyCode = this._buyNowCart ? braintreepaypal.currencyCode : state.config.getStoreConfigOrThrow().currency.code;
```

Comment:
Don't we validate above for `currencyCode` to be in `braintreepaypal` object?

Reply:
@animesh1987 Yeap
We validate it in `initialize` method for if this is BuyNow flow
But here we need to understand where we must get this value

Reply:
if the value is validated above, that should mean that we use the value in `braintreepaypal ` and don't need to have a fallback on display currency.

Reply:
@animesh1987 i have updated this point